services/app-api/utils/formTemplates/formTemplates.ts

```diff
@@ -89,9 +89,116 @@ export const nextTwelveQuarters = (
   return formFields;
 };
 
-export const expandRepeatedFields = (formFields: FormField[]) => {
+export const targetPopulationsByQuarterKeys = (
+  fieldId: string,
+  workPlanFieldData?: AnyObject,
+  workPlanMetaData?: AnyObject
+) => {
+  var targetPopulations = workPlanFieldData?.targetPopulations;
+  const keys = [];
+  for (let population of targetPopulations) {
+    keys.push([
+      workPlanMetaData?.reportPeriod,
+      fieldId,
+      population.transitionBenchmarks_targetPopulationName,
+    ]);
+  }
+  return keys;
+};
+
+export const targetPopulationsByQuarter = (
+  formFields: FormField[],
+  fieldToRepeat: FormField,
+  quarter: number,
+  workPlanFieldData?: AnyObject,
+  workPlanMetaData?: AnyObject
+) => {
+  var keys = targetPopulationsByQuarterKeys(
+    fieldToRepeat.id,
+    workPlanFieldData,
+    workPlanMetaData
+  );
+
+  if (quarter == 1) {
+    const contentString =
+      workPlanMetaData?.reportPeriod == 1
+        ? "First quarter (January 1 - March 31)"
+        : "Third quarter (July 1 - September 30)";
+    const formFieldHeader: FormField = {
+      id: `Period${workPlanMetaData?.reportPeriod}_Q1_header`,
+      type: "sectionHeader",
+      props: {
+        content: contentString,
+      },
+    };
+    formFields.push(formFieldHeader);
+  } else if (quarter == 2) {
+    const contentString =
+      workPlanMetaData?.reportPeriod == 1
+        ? "Second quarter (April 1 - June 30)"
+        : "Fourth quarter (October 1 - December 31)";
+    const formFieldHeader: FormField = {
+      id: `Period${workPlanMetaData?.reportPeriod}_Q2_header`,
+      type: "sectionHeader",
+      props: {
+        content: contentString,
+      },
+    };
+    formFields.push(formFieldHeader);
+  }
+
+  for (let key of keys) {
+    const formField: FormField = {
+      ...fieldToRepeat,
+      id: `Period${key[0]}_Q${quarter}_${key[2]}${key[2]}`,
+      type: fieldToRepeat?.type,
+      validation: fieldToRepeat.validation,
+      props: {
+        ...fieldToRepeat?.props,
+        label: `Number of ${key[2]}`,
+      },
+    };
+    formFields.push(formField);
+  }
+
+  return formFields;
+};
+
+export const targetPopulationsByReportingPeriod = (
+  formFields: FormField[],
+  fieldIndex: number,
+  fieldToRepeat: FormField,
+  workPlanFieldData?: AnyObject,
+  workPlanMetaData?: AnyObject
+) => {
+  targetPopulationsByQuarter(
+    formFields,
+    fieldToRepeat,
+    1,
+    workPlanFieldData,
+    workPlanMetaData
+  );
+
+  targetPopulationsByQuarter(
+    formFields,
+    fieldToRepeat,
+    2,
+    workPlanFieldData,
+    workPlanMetaData
+  );
```

Comment:
Nit: If this method is always called twice, with quarters 1 and 2, could you just call it once and have it do both?

Reply:
For now this method is only called twice, but for other pages in future implementation it'll be called a different amount of times, which is why I wanted to divide it up this way. I could put this in a loop so that there's less repeated code

Reply:
This clears things up. I'll leave it to you, just wanted to check!